app/src/main/java/app/revanced/integrations/boostforreddit/FixSLinksPatch.java

```diff
@@ -0,0 +1,93 @@
+package app.revanced.integrations.boostforreddit;
+
+import android.content.Context;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+import android.os.StrictMode;
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+import app.revanced.integrations.shared.Logger;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.SocketTimeoutException;
+import java.net.URL;
+
+public final class FixSLinksPatch {
+    public static String getUserAccessToken(Context context) {
+        File dbFile = context.getDatabasePath("reddit.db");
+        if (!dbFile.exists()) {
+            Logger.printInfo(() -> "Reddit database is not present at " + dbFile.getPath());
+            return null;
+        }
+        SQLiteDatabase redditDb = SQLiteDatabase.openDatabase(
+                dbFile.getPath(),
+                null, SQLiteDatabase.OPEN_READONLY);
+        Cursor tokenCursor = redditDb.rawQuery("SELECT AccessToken FROM User", null);
+        if (tokenCursor.getCount() == 0) {
+            Logger.printInfo(() -> "No authorized users found");
+            return null;
+        }
+        tokenCursor.moveToFirst();
+        String token = tokenCursor.getString(0);
+        tokenCursor.close();
+        redditDb.close();
+        return token;
+    }
+
+    public static String resolveSLink(Context context, String link) {
```

Comment:
Probably a good idea to introduce an interface class and use singletons to access the implementing methods via a patch since there's now two and possibly in the future, more resolve s link patches

Reply:
I am not particularly versed in java - but perhaps abstract class would make more sense? Considering that both patches right now do basically the same thing, only `getUserAccessToken` needs to be overridden.

Reply:
Checks out to me 👍